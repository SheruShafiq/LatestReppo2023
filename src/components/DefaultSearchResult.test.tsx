import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import DefaultSearchResult from "../components/DefaultSearchResult";
import matchers from "@testing-library/jest-dom";
import { debug } from "vitest-preview";
expect.extend(matchers);

test("renders DefaultSearchResult component correctly", () => {
  const theme = createTheme();
  const { getByText, getByAltText } = render(
    <ThemeProvider theme={theme}>
      <DefaultSearchResult />
    </ThemeProvider>
  );

  const imgElement = getByAltText("");
  expect(imgElement).toBeInTheDocument();
  expect(imgElement.src).toContain("undraw_search.svg");
  expect(imgElement.style.width).toBe("215px");

  const titleElement = getByText("Zoek binnen uw portefeuille");
  expect(titleElement).toBeInTheDocument();
  expect(titleElement.tagName).toBe("H1");

  expect(titleElement).toHaveStyle('text-align: center')
  expect(titleElement).toHaveStyle('margin-top: 2rem')

  const subtitleElement = getByText("Gebruik de zoekfunctie om klanten te vinden in uw portefeuille");
  expect(subtitleElement).toBeInTheDocument();
  expect(subtitleElement.tagName).toBe("P");
  expect(subtitleElement).toHaveStyle('text-align: center')
  expect(subtitleElement).toHaveStyle('margin-top: 1rem')
  expect(subtitleElement).toHaveStyle('margin-bottom: 2rem')
  expect(screen).toMatchSnapshot();
});
