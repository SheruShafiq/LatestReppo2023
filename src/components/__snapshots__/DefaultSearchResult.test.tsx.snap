// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders DefaultSearchResult component correctly 1`] = `Screen {}`;
